nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/assets/styles/_app.scss

```diff
@@ -127,186 +131,274 @@
-    $is-dark: map.get($color-config, is-dark);
+    $is-dark: map.get($material-theme-color-config, 'is-dark');
     $snack-surface: if($is-dark, #d9d9d9, #333333);
-    $snack-action-color: utils.get-color-on-surface($color-config, $snack-surface, 'accent');
+    // calculate the color for text on top of themed snackbar surface colored element
+    $snack-action-color: utils.get-color-on-surface($material-theme-color-config, $snack-surface, 'accent');
 
     * {
         // Tailwind sets a default that doesn't shift with light and dark themes
-        border-color: $on-surface-lighter;
         --mat-snack-bar-button-color: #{$snack-action-color};
     }
+}
 
-    .cdk-drag-disabled {
-        cursor: not-allowed !important;
-        background: $on-surface-lighter !important;
+@mixin generate-nifi-theme($nifi-theme) {
+    // Get the color config from the theme.
+    $nifi-theme-color-config: mat.get-color-config($nifi-theme);
+
+    // Get the color palettes from the color-config.
+    $nifi-theme-surface-palette: map.get($nifi-theme-color-config, 'primary');
+    $nifi-theme-success-palette: map.get($nifi-theme-color-config, 'accent');
+    $nifi-theme-warn-palette: map.get($nifi-theme-color-config, 'warn');
+
+    // default/lighter/darker colors from the nifi theme palettes
+    $nifi-theme-surface-palette-default: mat.get-color-from-palette($nifi-theme-surface-palette, 'default');
+    $nifi-theme-surface-palette-lighter: mat.get-color-from-palette($nifi-theme-surface-palette, 'lighter');
+    $nifi-theme-surface-palette-darker: mat.get-color-from-palette($nifi-theme-surface-palette, 'darker');
+    $nifi-theme-surface-palette-A200: mat.get-color-from-palette($nifi-theme-surface-palette, 'A200');
+    $nifi-theme-success-palette-default: mat.get-color-from-palette($nifi-theme-success-palette, 'default');
+    $nifi-theme-success-palette-lighter: mat.get-color-from-palette($nifi-theme-success-palette, 'lighter');
+    $nifi-theme-success-palette-darker: mat.get-color-from-palette($nifi-theme-success-palette, 'darker');
+    $nifi-theme-warn-palette-default: mat.get-color-from-palette($nifi-theme-warn-palette, 'default');
+    $nifi-theme-warn-palette-lighter: mat.get-color-from-palette($nifi-theme-warn-palette, 'lighter');
+    $nifi-theme-warn-palette-darker: mat.get-color-from-palette($nifi-theme-warn-palette, 'darker');
+
+    // semantic classes for default/lighter/darker colors for each palette in each theme
+    .surface-default {
+        color: utils.get-surface($nifi-theme-color-config) !important;
+        fill: utils.get-surface($nifi-theme-color-config) !important;
     }
 
-    .cdk-drag-preview {
-        box-sizing: border-box;
-        border-radius: 4px;
-        box-shadow: 0 3px 6px $canvas-primary-palette-A200;
+    .surface-lighter {
+        color: utils.get-surface($nifi-theme-color-config, lighter) !important;
+        fill: utils.get-surface($nifi-theme-color-config, lighter) !important;
```

Comment:
get-surface() doesn't have a mapping for 'lighter', it's going to return whatever mat.get-color-from-palette($color-palette, NULL); is

Reply:
oh yea my bad. Good catch.